db/migrate/20221010090227_add_devise_two_factor_to_users.rb

```diff
@@ -0,0 +1,9 @@
+class AddDeviseTwoFactorToUsers < ActiveRecord::Migration[6.1]
+  def change
+    add_column :users, :encrypted_otp_secret, :string
+    add_column :users, :encrypted_otp_secret_iv, :string
+    add_column :users, :encrypted_otp_secret_salt, :string
+    add_column :users, :consumed_timestep, :integer
```

Comment:
is `consumed_timestep` being used?

Reply:
No, it is not being used but according to : https://github.com/tinfoil/devise-two-factor/blob/v4.x/UPGRADING.md , I believe we might need it: "Pull request #43 added a new field to protect against "shoulder-surfing" attacks. If upgrading, you'll need to add the :consumed_timestep column to your Users model."